### PR TITLE
Handle invalid curve points

### PR DESCRIPTION
**Pull request by andrewwhitehead**
Also replaces some invalid test data.

### CODE REVIEW

File: src/amcl.rs [hunk 1]
```diff
@@ -649,6 +673,111 @@ impl<'a> Deserialize<'a> for Pair {
...
+    rhs.reduce();
+    rhs.sqr();
+    rhs.mul(&x);
+    lhs.sub(&rhs);
+    rhs.copy(&z);
+    rhs.reduce();
+    rhs.sqr();
+    rhs.mul(&z);
+    rhs.dbl(); // b = 2
+    lhs.equals(&rhs) || point.is_infinity()
```
[the remark this answers is not in the document]
Comment (andrewwhitehead) [reply src/amcl.rs:736]: It's not being checked here, no. It should likely be checked at the point of usage, although we could also just ban all infinity points if they aren't needed.

[the remark this answers is not in the document]
Comment (andrewwhitehead) [reply src/amcl.rs]: I changed it to reject infinity points, which works with the test data, and real world testing can see if that needs to be updated.

**Comment by swcurran** [comment]
@mikelodder7 — are you happy with the solution @andrewwhitehead has implemented — banning infinity points?

**Review by blu3beri — APPROVED** [review]
Approved for @mikelodder7

**Pull request merged**